supervision/detection/line_counter.py

```diff
@@ -26,16 +26,61 @@ class LineZone:
             to outside.
     """
 
-    def __init__(self, start: Point, end: Point):
+    def __init__(self, start: Point, end: Point, count_condition="whole_crossed"):
```

Comment:
Here's an idea. Let's replace `count_condition` with `triggering_anchors: List[sv.Position]`. In this way, we can provide our users with full flexibility. 

Let me know if you need more explanation. :)

Reply:
I believe I implemented just what you needed. Please check so we can discuss further if need be.